scipy/stats/_distn_infrastructure.py

```diff
@@ -1540,6 +1541,23 @@ def _penalized_nnlf(self, theta, x):
         n_log_scale = len(x) * log(scale)
         return self._nnlf_and_penalty(x, args) + n_log_scale
 
+    def _get_shape_info(self):
+        return self._shape_info()
+
+
+class ShapeInfo:
+    def __init__(self, shape_name, integrality=False, domain=(-np.inf, np.inf),
+                 inclusive=(True, True)):
```

Comment:
I am still wondering if we should have a single container for all shape parameters or like you did with a list. All params could also be vectors to represent all shapes. Anyway we can start like that.